src/common/transformations/src/transformations/op_conversions/convert_interpolate11_downgrade.cpp

```diff
@@ -0,0 +1,72 @@
+
+    const auto interpolate_v11_pattern = pattern::wrap_type<opset11::Interpolate>();
+
+    const matcher_pass_callback callback = [=](pattern::Matcher& m) {
+        const auto v4_compatible_interpolation_mode = [](const op::util::InterpolateBase::InterpolateMode mode) {
+            constexpr std::array<op::util::InterpolateBase::InterpolateMode, 4> allowed_modes = {
+                op::util::InterpolateBase::InterpolateMode::NEAREST,
+                op::util::InterpolateBase::InterpolateMode::LINEAR,
+                op::util::InterpolateBase::InterpolateMode::LINEAR_ONNX,
+                op::util::InterpolateBase::InterpolateMode::CUBIC};
+
+            return std::find(std::begin(allowed_modes), std::end(allowed_modes), mode) != std::end(allowed_modes);
+        };
+
+        const auto interpolate_v11 = std::dynamic_pointer_cast<opset11::Interpolate>(m.get_match_root());
+        if (!interpolate_v11 || !v4_compatible_interpolation_mode(interpolate_v11->get_attrs().mode) ||
+            transformation_callback(interpolate_v11)) {
+            return false;
+        }
+
+        // downgrade only if the interpolation mode used to create v11 is supported by v4
+        std::shared_ptr<ov::opset4::Interpolate> interpolate_v4;
+        ov::Output<ov::Node> v4_input_output_shape;
+        ov::Output<ov::Node> v4_input_scales;
+
+        if (interpolate_v11->get_attrs().shape_calculation_mode ==
+            ov::op::util::InterpolateBase::ShapeCalcMode::SCALES) {
+            v4_input_scales = interpolate_v11->input_value(1);
+            v4_input_output_shape = opset4::Constant::create(element::i32, Shape{}, {1});
+        } else {
+            v4_input_output_shape = interpolate_v11->input_value(1);
+            v4_input_scales = opset4::Constant::create(element::f32, Shape{}, {1.0f});
+        }
+
+        if (interpolate_v11->get_input_size() == 3) {  // with axes input
+            interpolate_v4 = std::make_shared<ov::opset4::Interpolate>(interpolate_v11->input_value(0),
+                                                                       v4_input_output_shape,
+                                                                       v4_input_scales,
+                                                                       interpolate_v11->input_value(2),
+                                                                       interpolate_v11->get_attrs());
+        } else {
+            interpolate_v4 = std::make_shared<ov::opset4::Interpolate>(interpolate_v11->input_value(0),
+                                                                       v4_input_output_shape,
+                                                                       v4_input_scales,
+                                                                       interpolate_v11->get_attrs());
+        }
+
+        interpolate_v4->set_friendly_name(interpolate_v11->get_friendly_name());
+        copy_runtime_info(interpolate_v11, interpolate_v4);
```

Comment:
I would recommend to copy runtime_info for all new nodes:
`copy_runtime_info(interpolate_v11, {interpolate_v4, v4_input_scales, v4_input_output_shape);`
because these 3 nodes are actually a decomposition of interpolate_v11 using the prev interpolate version.
Not sure if anyone uses this information, but this is correct.

Reply:
Agree, thank you for the comment

Reply:
I didn't notice that, the correct version looks good